.github/workflows/unit-tests-pr.yml

```diff
@@ -0,0 +1,25 @@
+name: "Unit Tests"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - ready_for_review
+      - reopened
+  pull_request_review:
+    types:
+      - submitted
+      - edited
+
+concurrency:
+  group: unit-tests-${{ github.event.number }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Unit Tests
+    if: ${{ github.event.review.state == 'APPROVED' || github.event_name == 'pull_request' }}
```

Comment:
this is the way to check if the review was an approve as per https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved